tests/engine/test_process.py

```diff
@@ -469,3 +469,25 @@ def test_input_validation_storable_nodes():
     """
     with pytest.raises(ValueError):
         run(TestValidateDynamicNamespaceProcess, **{'namespace': {'a': 1}})
+
+
+class TestNotRequiredNoneProcess(Process):
+    """Process with an optional input port that should therefore also accept ``None``."""
+
+    _node_class = orm.WorkflowNode
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+        spec.input('not_required', valid_type=orm.Int, required=False)
+
+
+@pytest.mark.usefixtures('clear_database_before_test')
+def test_not_required_accepts_none():
+    """Test that a port that is not required, accepts ``None``."""
+    from aiida.engine.utils import instantiate_process
+    from aiida.manage import get_manager
+
+    runner = get_manager().get_runner()
+    process = instantiate_process(runner, TestNotRequiredNoneProcess, not_required=None)
```

Comment:
I think the process function with input parameter defaulting to `None` is not covered by this test, correct? Can you add to nightly tests which are exactly the process function manifest the case?

Reply:
That is already being tested: https://github.com/aiidateam/aiida-core/blob/9c54b73da1664eab0279e323d3372fcf6102dcae/tests/engine/test_process_function.py#L244
Here we are adding the same behavior for any other process, such as `CalcJob` and `WorkChain`.

Reply:
Ah, I see. The `FunctionProcess` is subclass of `Process` I thought this PR is going to fix the bug introduced by the previously added feature for `None` supported input. 
It is all good to me! Thanks!